### PR TITLE
Aut 30/add roles to common passwords dynamodb

### PR DESCRIPTION
**Pull request by DTaylorGDS**
## What?

Add read permissions for new common passwords Dynamo table for three lambdas, which will use this to validate whether or not passwords is a common password and produce an error accordingly.

## Why?

So that password validation can consider common words

## Related PRs

https://github.com/alphagov/di-authentication-api/pull/1990

### CODE REVIEW

File: ci/terraform/oidc/dynamo-policies.tf [hunk 1]
```diff
@@ -30,7 +34,7 @@ data "aws_iam_policy_document" "dynamo_user_write_policy_document" {
     ]
     resources = [
       data.aws_dynamodb_table.user_credentials_table.arn,
-      data.aws_dynamodb_table.user_profile_table.arn,
+      data.aws_dynamodb_table.common_passwords_table.arn,
```
Comment (JHjava) [inline ci/terraform/oidc/dynamo-policies.tf:37]: Is this change intentional?

**Review by JHjava — COMMENTED** [review]
PR looks great. Just one question.

File: ci/terraform/oidc/dynamo-policies.tf (hunk 1 above)
Comment (DTaylorGDS) [reply ci/terraform/oidc/dynamo-policies.tf]: Thanks for pointing out - it was accidental on my part.

**Review by JHjava — APPROVED** [review]
